xarray/core/coordinates.py

```diff
@@ -482,3 +800,60 @@ def assert_coordinate_consistent(
                 f"dimension coordinate {k!r} conflicts between "
                 f"indexed and indexing objects:\n{obj[k]}\nvs.\n{coords[k]}"
             )
+
+
+def create_coords_with_default_indexes(
+    coords: Mapping[Any, Any], data_vars: Mapping[Any, Variable] | None = None
+) -> Coordinates:
+    """Maybe create default indexes from a mapping of coordinates."""
+
+    # Note: data_vars are needed here only because a pd.MultiIndex object
+    # can be promoted as coordinates.
+    # TODO: It won't be relevant anymore when this behavior will be dropped
+    # in favor of the more explicit ``Coordinates.from_pandas_multiindex()``.
+
+    from xarray.core.dataarray import DataArray
+
+    all_variables = dict(coords)
+    if data_vars is not None:
+        all_variables.update(data_vars)
+
+    indexes: dict[Hashable, Index] = {}
+    variables: dict[Hashable, Variable] = {}
+
+    # this is needed for backward compatibility: when a pandas multi-index
+    # is given as data variable, it is promoted as index / level coordinates
```

Comment:
Can we issue a FutureWarning in this case, to note that MultiIndex should be supplied in coordinates?

Reply:
I was thinking adding it in a follow-up PR (as well as for any MultiIndex given as coordinate data), but yes we can do it here.